bsp/worker/src/mill/bsp/worker/MillBuildServer.scala

```diff
@@ -105,7 +54,7 @@ class MillBuildServer(
             val modules: Seq[Module] =
               ModuleUtils.transitiveModules(evaluator.rootModule) ++ Seq(`mill-build`)
             val map = modules.collect {
-              case m: MillBuildModule =>
+              case m: MillBuildRootModule =>
```

Comment:
I wonder whether this is still a singleton. Do we currently project meta-builds to individual synthetic modules? If not, should we? 

I think, we should, if possible. To provide a nicer editing experience. On each level we might have different dependencies and settings, so separating them as (synthetic) modules make sense. But if we plan to do that, we should not match simply match on the type here, as we assume hard-coded module segments in the code below.

Reply:
@lefou no, it should no longer be a singleton. However, this PR assumes it is, basically ignoring meta-builds

It shouldn't be too hard to make it a non-singleton, we'd just need to move `MIllBuildBootstrap` over to make it available, and then fish out the `MillBuildRootModule` from each level.

Reply:
I sketched out what it would look like handling all the meta-module's `MillBuildRootModule`, rather than just hardcoding one. But I'll probably need your help to test this and poke and it to get it working

Reply:
I currently just run vscode (which I'm not using myself regularily) on a recent Mill project and test a bit manually, but that's unsatisfying and way too ineffective. There are some BSP test suites out there which we could use, also a recently renewed one (https://github.com/build-server-protocol/bsp-testkit2), but they all come with no documentation and I simply lack the time to dig deeper how to use/integrate them.